File: app/src/main/java/com/originb/inkwisenote/modules/smartnotes/ui/SmartNotebookPageScrollLayout.java
package com.originb.inkwisenote.modules.smartnotes.ui;

import android.content.Context;
import androidx.recyclerview.widget.LinearLayoutManager;
import lombok.Setter;

public class SmartNotebookPageScrollLayout extends LinearLayoutManager {

    @Setter
    private boolean scrollRequested;

    public SmartNotebookPageScrollLayout(Context context) {
        super(context, LinearLayoutManager.HORIZONTAL, false);
        scrollRequested = false;
    }

    @Override
    public boolean canScrollVertically() {
        return false;
    }

    @Override
    public boolean canScrollHorizontally() {
        return scrollRequested;
    }
}

File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/ui/SmartNotebookPageScrollLayout.kt
package com.originb.inkwisenote2.modules.smartnotes.ui

import android.content.Context
import androidx.recyclerview.widget.LinearLayoutManager
class SmartNotebookPageScrollLayout(context: Context?) : LinearLayoutManager(context, HORIZONTAL, false) {
    private var scrollRequested = false

    fun setScrollRequested(requested: Boolean) {
        scrollRequested = requested
    }

    override fun canScrollVertically(): Boolean {
        return false
    }

    override fun canScrollHorizontally(): Boolean {
        return scrollRequested
    }
}
